frontend/src/metabase-lib/limit.ts

```diff
@@ -1,15 +1,21 @@
 import * as ML from "cljs/metabase.lib.limit";
 import type { Query, Limit } from "./types";
 
-export function currentLimit(query: Query): Limit {
-  return ML.current_limit(query);
-}
+const DEFAULT_STAGE_INDEX = -1;
 
-export function limit(query: Query, limit: Limit): Query {
-  return ML.limit(query, limit);
+export function currentLimit(
+  query: Query,
+  stageIndex = DEFAULT_STAGE_INDEX,
+): Limit {
+  return ML.current_limit(query, stageIndex);
 }
 
-export function hasLimit(query: Query) {
-  const limit = currentLimit(query);
+declare function LimitFn(query: Query, limit: Limit): Query;
+declare function LimitFn(query: Query, stageIndex: number, limit: Limit): Query;
+
+export const limit: typeof LimitFn = ML.limit;
```

Comment:
This allows using `limit` with or without a query stage index provided:

```ts
Lib.limit(query, 5) // Set query limit to 5
Lib.limit(query, 2, 5) // Set limit to 5 for a query at stage 2
```

Using an optional parameter is not great here for two reasons:

1. We can't have an optional argument between two required args, and a signature `query, limit, stage` looks odd
2. When passed to Clojure, `undefined` is going to be treated like `nil`, and MLv2 doesn't handle it well yet